apps/fabric-website/src/components/Nav/Nav.tsx

```diff
@@ -53,23 +66,74 @@ export class Nav extends React.Component<INavProps, INavState> {
 
         {childLinks}
       </li>
+    </span>
     );
   }
+ 
+  private _getSearchBox(val) {
+    if (val === 'Components') {
+      return (
+        <div style={{ display: 'flex' }}>
+          <SearchBox
+            placeholder="Search"
+            underlined={true}
+            styles={{
+              iconContainer: {
+                color: 'white'
+              },
+              icon: {
+                color: 'white'
+              },
+              clearButton: {
+                selectors: {
+                  '.ms-Button': {
+                    color: 'white'
+                  }
+                }
+              },
+              field: {
+                backgroundColor: 'transparent',
+                color: 'white'
+              },
+              root: {
+                marginBottom: '5px',
+                width: '180px'
+              }
+            }}
+            onChange={this._onChangeQuery.bind(this)}
+          />
+        </div>
+      );
+    }
+  }
+
+   private _renderLinkList(pages: INavPage[], isSubMenu: boolean): React.ReactElement<{}> {
+    let { searchQuery } = this.state;
 
-  private _renderLinkList(pages: INavPage[], isSubMenu: boolean): React.ReactElement<{}> {
     const links: React.ReactElement<{}>[] = pages
       .filter(page => !page.hasOwnProperty('isHiddenFromMainNav'))
-      .map((page: INavPage, linkIndex: number) => this._renderLink(page, linkIndex));
-
-    return (
-      <ul className={css(styles.links, isSubMenu ? styles.isSubMenu : '')} aria-label="Main website navigation">
-        {links}
-      </ul>
-    );
+      .filter(
+        page =>
+          page.title.toLowerCase().indexOf(searchQuery.toLowerCase()) > -1 ||
+          page.url.toLowerCase().indexOf('#/components/')
```

Comment:
indexOf(...) > -1, right?

Reply:
or do you really mean indexOf() == 0 ?

Reply:
it only works like that, ==0 and > -1 hides links that are not part of the components list

Reply:
Then it should be >0